src/backend/gpopt/utils/COptTasks.cpp

```diff
@@ -498,7 +499,10 @@ COptTasks::OptimizeTask(void *ptr)
 		CMDCache::Init();
 		CMDCache::SetCacheQuota(optimizer_mdcache_size * 1024L);
 	}
-	else if (reset_mdcache)
+	else if (reset_mdcache ||
+			 (CMDCache::IsTransient() &&
+			  !TransactionIdEquals(TransactionXmin,
+								   CMDCache::GetTransientXmin())))
```

Comment:
1. Write appropriate test on invalidation of cached plan.
2.  Do we need both `bool m_transient` and `m_transientXmin` variables at the same time

Reply:
I can add EXPLAIN after the transaction is end to create_index_hot.spec or should I look for another place?

Reply:
> I can add EXPLAIN after the transaction is end to create_index_hot.spec or should I look for another place?

Well, we can't write the regress tests, and about isolation tests, I can't any isolation test with the explain output. Therefore, we can omit this, I guess.